cpp/unit/unit.cpp

```diff
@@ -106,16 +107,20 @@ void Unit::update_secondary(int64_t time_elapsed) {
 }
 
 void Unit::draw() {
+	this->draw(this->location, this->graphics);
```

Comment:
if you're worried about the performance impact or anything, just `#include "util/compiler.h"` add `unlikely()` around the check condition; that should make the check pretty much zero-cost.